### PR TITLE
pythonPackages.pytest-instafail: init at 0.4.2

### PR DESCRIPTION
**Pull request by jacg**
###### Motivation for this change

Package not yet available in nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 110138` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.pytest-instafail</li>
    <li>python38Packages.pytest-instafail</li>
    <li>python39Packages.pytest-instafail</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Please do not rebase PRs on master when not fixing merge conflicts or there is a good reason for it.

File: pkgs/development/python-modules/pytest-instafail/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "pytest-instafail";
+  version = "0.4.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "10lpr6mjcinabqynj6v85bvb1xmapnhqmg50nys1r6hg7zgky9qr";
+  };
+
+  checkInputs = [ pytestCheckHook ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-instafail/default.nix:17]:
```suggestion
  pythonImportsCheck = [ "pytest_instafail" ];
```

see https://nixos.org/manual/nixpkgs/stable/#python

**Review by fabaff — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110138` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.pytest-instafail</li>
<li>python38Packages.pytest-instafail</li>
<li>python39Packages.pytest-instafail</li>
</ul>
</details>

File: pkgs/development/python-modules/pytest-instafail/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  meta = {
+    description = "pytest plugin that shows failures and errors instantly instead of waiting until the end of test session";
+    homepage = "https://github.com/pytest-dev/pytest-instafail";
+    license = lib.licenses.free;
```
[the remark this answers is not in the document]
Comment (jacg) [reply pkgs/development/python-modules/pytest-instafail/default.nix]: Hmm, indeed. I was fooled by the fact that "BSD" doesn't appear anywhere in the licence file.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110138` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.pytest-instafail</li>
<li>python38Packages.pytest-instafail</li>
<li>python39Packages.pytest-instafail</li>
</ul>
</details>

**Pull request merged**